lib/device_model.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:fwupd/fwupd.dart';
+import 'package:safe_change_notifier/safe_change_notifier.dart';
+
+import 'firmware_model.dart';
+
+class DeviceModel extends SafeChangeNotifier {
+  DeviceModel(this._firmwareModel, this._device)
```

Comment:
it's a matter of taste but it would somehow feel more politically correct to pass the list of releases and the service to operate on, instead of another model. what do you think?

Reply:
let's skip this idea for now! we can improve the structure with proxy providers as a separate refactoring step later